### PR TITLE
Testeng 1871 Add SendBuildEvent to release pipeline

### PR DESCRIPTION
**Pull request by matthew-wiemer**
Successful build (same code, run in featurebranch.yaml): https://github.com/relativitydev/relativity.testing.framework/actions/runs/1607616017
Result in NewRelic: https://onenr.io/08dQelmNDje

Complete the following tasks before merging in your PR:
- [x] If this is a user facing change, update [version.txt](https://github.com/relativitydev/relativity.testing.framework/blob/master/version.txt) and [CHANGELOG.md](https://github.com/relativitydev/relativity.testing.framework/blob/master/CHANGELOG.md)

### CODE REVIEW

File: .github/workflows/ReleaseBranch.yaml [hunk 1]
```diff
@@ -136,7 +136,54 @@ jobs:
...
+        ClientSecret: ${{ secrets.CLIENTSECRET }}
+      run: |
+        $Body = @{
+          grant_type = "client_credentials"
+          client_id = "${{ env.ClientId }}"
+          client_secret = "${{ env.CLIENTSECRET }}"
+          scope = "https://vault.azure.net/.default"
+        }
+
+        $AccessToken = (Invoke-RestMethod -Method POST `
```
Comment (paulletang) [inline .github/workflows/ReleaseBranch.yaml:154]: What was the fix for the not being able to access the secrets?

File: .github/workflows/ReleaseBranch.yaml [hunk 2]
```diff
@@ -136,7 +136,54 @@ jobs:
...
+        $Merge = git show -s --format=%ci "${{ github.sha }}"
+        $Now = Get-Date
+        $Duration = (New-TimeSpan -Start $Merge -End $Now).Seconds
+
+        # Of 'myorg/reponame' take reponame
+        $ProductName = ("${{ github.repository }}").split("/")[1]
+
+        $BuildEventPayload = "{
+            `"event`": true,
+            `"name`": `"buildtest`",
```
Comment (paulletang) [inline .github/workflows/ReleaseBranch.yaml:171]: This should be `build`, right?

File: .github/workflows/ReleaseBranch.yaml [hunk 3]
```diff
@@ -136,7 +136,54 @@ jobs:
...
+            `"job.sub_id`": `"${{ github.run_attempt }}`",
+            `"job.result`": `"${{ job.status }}`",
+            `"job.duration`": $Duration,
+            `"job.name`": `"${{ github.repository }}`",
+            `"job.url`": `"${{ github.server_url }}/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}`",
+            `"application.name`": `"$ProductName`",
+            `"type`": `"devops_metrics`"
+        }"
+
+        Write-Output $BuildEventPayload
```
Comment (paulletang) [inline .github/workflows/ReleaseBranch.yaml:183]: Do you have a sample `$BuildEventPayload` to show the output of?

File: .github/workflows/ReleaseBranch.yaml (hunk 1 above)
Comment (matthew-wiemer) [reply .github/workflows/ReleaseBranch.yaml:154]: When they were rolled, it worked. My conclusion is that the values configured in Github had expired (in Keyvault). I should have explicitly sanity-checked that the configured secret was the latest version.

https://kcura-pd.slack.com/archives/C02JQ9874Q7/p1639520126091300

File: .github/workflows/ReleaseBranch.yaml (hunk 2 above)
Comment (matthew-wiemer) [reply .github/workflows/ReleaseBranch.yaml:171]: Yes, will fix.

File: .github/workflows/ReleaseBranch.yaml (hunk 3 above)
Comment (matthew-wiemer) [reply .github/workflows/ReleaseBranch.yaml:183]: Here: https://github.com/relativitydev/relativity.testing.framework/runs/4587338434?check_suite_focus=true is a sample 

```
{
    "event": true,
    "name": "buildtest",
    "r1.team.id": "PTCI-1599",
    "job.id": "1603869152",
    "job.sub_id": "1",
    "job.result": "success",
    "job.duration": 0,
    "job.name": "",
    "job.url": "/runs//attempts/",
    "application.name": "",
    "type": "devops_metrics"
}
```

Comment (matthew-wiemer) [reply .github/workflows/ReleaseBranch.yaml:183]: Thought I removed this from the release code though, now I'm thinking it doesn't hurt to have in around - thoughts?

Comment (briantoms) [reply .github/workflows/ReleaseBranch.yaml:183]: That `job.url` does not look right. Seems like a couple of values are missing.

Comment (matthew-wiemer) [reply .github/workflows/ReleaseBranch.yaml:183]: gah, that's an old one - my mistake. here's a more current one:  https://github.com/relativitydev/relativity.testing.framework/runs/4597129547?check_suite_focus=true

```
{
    "event": true,
    "name": "buildtest",
    "r1.team.id": "PTCI-1599",
    "job.id": "1607616017",
    "job.sub_id": "1",
    "job.result": "success",
    "job.duration": 34,
    "job.name": "relativitydev/relativity.testing.framework",
    "job.url": "https://github.com/runs/1607616017/attempts/1",
    "application.name": "relativity.testing.framework",
    "type": "devops_metrics"
}
```

Comment (briantoms) [reply .github/workflows/ReleaseBranch.yaml:183]: Is it expected that the URL does not resolve?

**Review by briantoms — APPROVED** [review]

**Pull request merged**